apps/server/src/migrations/migrations.ts

```diff
@@ -6,6 +6,66 @@
 
 // Migrations should be kept in descending order, so the latest migration is first.
 const MIGRATIONS: (SqlMigration | JsMigration)[] = [
+    // Add OCR results table for storing extracted text from images
+    {
+        version: 233,
+        sql: /*sql*/`\
+            -- Create OCR results table to store extracted text from images
+            CREATE TABLE IF NOT EXISTS ocr_results (
+                id INTEGER PRIMARY KEY AUTOINCREMENT,
+                entity_id TEXT NOT NULL,
+                entity_type TEXT NOT NULL DEFAULT 'note',
+                extracted_text TEXT NOT NULL,
+                confidence REAL NOT NULL,
+                language TEXT NOT NULL DEFAULT 'eng',
+                extracted_at TEXT NOT NULL,
+                created_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                updated_at TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                UNIQUE(entity_id, entity_type)
+            );
+
+            -- Create indexes for better search performance
+            CREATE INDEX IF NOT EXISTS idx_ocr_results_entity 
+            ON ocr_results (entity_id, entity_type);
+
+            CREATE INDEX IF NOT EXISTS idx_ocr_results_text 
+            ON ocr_results (extracted_text);
+
+            CREATE INDEX IF NOT EXISTS idx_ocr_results_confidence 
+            ON ocr_results (confidence);
+
+            -- Create full-text search index for extracted text
```

Comment:
It's interesting but I suppose this creates additional complexity. We don't yet have full-text search for note content.

So I think we should try to find a solution (doesn't have to be in the same PR) that allows full text search on the entire `blobs` table.